internal/provider/auth_aws.go

```diff
@@ -0,0 +1,186 @@
+specified at all and if reauthentication is allowed, then the method will generate a random nonce, 
+attaches it to the instance's identity-accesslist entry and returns the nonce back as part of auth metadata.
+This value should be used with further login requests, to establish client authenticity.
+Clients can choose to set a custom nonce if preferred, in which case, 
+it is recommended that clients provide a strong nonce.
+If a nonce is provided but with an empty value, 
+it indicates intent to disable reauthentication. 
+Note that, when disallow_reauthentication option is enabled on either the role or the role tag, 
+the nonce holds no significance. This is ignored unless using the ec2 auth method.`,
+			Optional: true,
+		},
+		consts.FieldIAMHttpRequestMethod: {
+			Type: schema.TypeString,
+			Description: `HTTP method used in the signed request.
+Currently only POST is supported, but other methods may be supported in the future.
+This is required when using the iam auth method.`,
+			Optional: true,
+		},
+		consts.FieldIAMHttpRequestURL: {
+			Type: schema.TypeString,
+			Description: `base64-encoded HTTP URL used in the signed request. 
+Most likely just aHR0cHM6Ly9zdHMuYW1hem9uYXdzLmNvbS8= (base64-encoding of https://sts.amazonaws.
+com/) as most requests will probably use POST with an empty URI. 
+This is required when using the iam auth method.`,
+			Optional: true,
+		},
+		consts.FieldIAMHttpRequestBody: {
+			Type: schema.TypeString,
+			Description: `base64-encoded body of the signed request. 
+Most likely QWN0aW9uPUdldENhbGxlcklkZW50aXR5JlZlcnNpb249MjAxMS0wNi0xNQ==, 
+which is the base64 encoding of Action=GetCallerIdentity&Version=2011-06-15.
+This is required when using the iam auth method.`,
+			Optional: true,
+		},
+		consts.FieldIAMHttpRequestHeaders: {
+			Type:     schema.TypeMap,
+			Optional: true,
+			Description: `key/value pairs of headers for use in the sts:GetCallerIdentity HTTP requests headers. 
+Can be either a Base64-encoded, JSON-serialized string, or a JSON object of key/value pairs. 
+The JSON serialization assumes that each header key maps to either a string value or an array of string values 
+(though the length of that array will probably only be one). 
+If the iam_server_id_header_value is configured in Vault for the aws auth mount, 
+then the headers must include the X-Vault-AWS-IAM-Server-ID header, 
+its value must match the value configured, and the header must be included in the signed headers. 
+This is required when using the iam auth method.`,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+	},
+})
+
+// AuthLoginAWS for handling the Vault AWS authentication engine.
+type AuthLoginAWS struct {
+	AuthLoginCommon
+}
+
+func (l *AuthLoginAWS) Schema() *schema.Resource {
+	return SchemaLoginAWS
+}
+
+func (l *AuthLoginAWS) MountPath() string {
+	if l.mount == "" {
+		return consts.AuthMethodUserpass
```

Comment:
should this be `consts.AuthMethodAWS`?

Reply:
Doh! Yes.

Reply:
Fixed in 640f974bad593f178a026097b5acb997608283d9